controllers/openstackdataplanenode_controller.go

```diff
@@ -94,6 +96,20 @@ func (r *OpenStackDataPlaneNodeReconciler) Reconcile(ctx context.Context, req ct
 		return ctrl.Result{}, err
 	}
 
+	_, result, err = ensureSecret(
```

Comment:
So we'll ensure the secret meets our constraints or throw an error here before the execution below.

Reply:
yes!
it will ensure the secret contains `ssh-privatekey`